### PR TITLE
Fix reading body failing with nil error

### PR DESCRIPTION
**Pull request by mshibuya**
This PR aims to fix the issue #58, which was caused by #51.

@arkadiyt Please review this, many users are facing this issue.
https://github.com/carrierwaveuploader/carrierwave/issues/2628

### CODE REVIEW

File: lib/ssrf_filter/ssrf_filter.rb [hunk 1]
```diff
@@ -191,17 +191,17 @@ def self.fetch_once(uri, ip, verb, options, &block)
...
-        http.request(request) do |response|
-          case response
-          when ::Net::HTTPRedirection
-            url = response['location']
-            # Handle relative redirects
-            url = "#{uri.scheme}://#{hostname}:#{uri.port}#{url}" if url.start_with?('/')
-            return nil, url
-          else
-            block&.call(response)
-            return response, nil
```
Comment (mshibuya) [inline lib/ssrf_filter/ssrf_filter.rb:203]: This use of using `return` within a block is the cause of the issue. It causes the execution to jump out from the method `#fetch_once` (not from this block), skipping [post-processing of body read](https://github.com/ruby/net-http/blob/65aed40f350f8d13f727221fb391c677688cb705/lib/net/http/response.rb#L186) by net-http. So the result is attempting to read the body afterwards, when `@socket` already cleared up with `nil`.

**Comment by niborg** [comment]
@arkadiyt any input on this?

**Comment by zumkorn** [comment]
@arkadiyt can you check this, please?

**Comment by basex** [comment]
@arkadiyt if you have some time, merge please this MR and release a new version. It's needed for projects with ruby 3.2

Versions previous to 1.1.1 won't work in ruby 3.2. https://github.com/arkadiyt/ssrf_filter/pull/54 fixed the problem.
Version 1.1.1 doesn't work well due to the issue https://github.com/arkadiyt/ssrf_filter/issues/58. This MR fixes the issue.

**Review by shoe — APPROVED** [review]

**Review by arkadiyt — APPROVED** [review]

**Comment by arkadiyt** [comment]
Sorry on the very long delay here, I'm getting back to this gem and going to maintain it more actively. Appreciate the fix

**Pull request merged**